src/array.ts

```diff
@@ -82,9 +82,7 @@ export class ArrayType<T extends IAnyType> extends BaseType<Array<T["InputType"]
   instantiate(snapshot: this["InputType"] | undefined, context: TreeContext, parent: IStateTreeNode | null): this["InstanceType"] {
     const array = new QuickArray<T>(this, parent, context);
     if (snapshot) {
-      for (let index = 0; index < snapshot?.length; ++index) {
-        array.push(this.childrenType.instantiate(snapshot[index], context, array));
-      }
+      array.push(...snapshot.map((element) => this.childrenType.instantiate(element, context, array)));
```

Comment:
It may be even faster to do this:

```js
if(snapshot) {
  return new QuickArray<T>(
    this,
    parent,
    context,
    ...snapshot.map((element) => this.childrenType.instantiate(element, context, array)
  );
}

return new QuickArray<T>(this, parent, context);
```

Reply:
I was hoping to do that as well but the child `instantiate` call needs a reference to the parent `array` object as the children are instantiated, and I wasn't really sure how to work around that :( I agree though it'd be much more preferrable. We could potentially set things up to set the parent value later but that seems kinda complicated and maybe not faster?

Reply:
Oh yeah, missed that parent reference 🤦 

Agreed that it'd be complicated to make that work, so let's worry about optimizing further when we've exhausted all other avenues 🚀

Reply:
I too launched right into doing it that way only to realize lol. Onwards!